braindecode/datasets/base.py

```diff
@@ -60,13 +62,25 @@ class WindowsDataset(BaseDataset):
     def __init__(self, windows, description):
         self.windows = windows
         self.description = description
+        self.y = np.array(self.windows.metadata.loc[:,'target'])
+        self.crop_inds = np.array(self.windows.metadata.loc[:,
+                              ['i_supercrop_in_trial', 'i_start_in_trial',
+                               'i_stop_in_trial']])
 
     def __getitem__(self, index):
-        x = self.windows.get_data(item=index)[0].astype('float32')
-        md = self.windows.metadata.iloc[index]
-        return x, md['target'], md[
-            ['i_supercrop_in_trial', 'i_start_in_trial',
-             'i_stop_in_trial']].to_list()
```

Comment:
I suspect it was
```
md[
            ['i_supercrop_in_trial', 'i_start_in_trial',
             'i_stop_in_trial']].to_list()
```
that was killing you

Reply:
I checked that already the call `md = self.windows.metadata.iloc[index]` was causing 3x slowdown without subsequent lines. Subsequent lines caused further slowdown...

Reply:
to be more precise, iterating through train set without md line:
`87.7 ms ± 343 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)`
adding the md line without using it:
`243 ms ± 2.95 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)`
pandas version '1.0.1'